error-prone-contrib/src/main/java/tech/picnic/errorprone/bugpatterns/util/SourceCode.java

```diff
@@ -38,10 +38,42 @@ private SourceCode() {}
    * @return A non-{@code null} string.
    */
   public static String treeToString(Tree tree, VisitorState state) {
+    /** The issue starts inside `state.getSourceForNode(tree)` statement because line:
+     * CharSequence source = getSourceCode()
+     *
+     * returns =>
+     * """
+     *import lombok.Data;
+     *
+     * @Data
+     * public class AnyClass {
+     *  private String anything;
+     * }
+     * """
+     * Which is not the correct start & end position of the intended replacement statement.
+
+      if (end == -1) {
+        return null;
+      }
+      checkArgument(start >= 0, "invalid start position (%s) for: %s", start, tree);
+      checkArgument(start < end, "invalid source positions (%s, %s) for: %s", start, end, tree);
+      checkArgument(end <= source.length(), "invalid end position (%s) for: %s", end, tree);
+      return source.subSequence(start, end).toString();
+     */
     String src = state.getSourceForNode(tree);
     return src != null ? src : tree.toString();
   }
 
+
+
+  /**
+   * TODO: Call this conditionally from DirectReturn ??
```

Comment:
This will not help since the correctly replaced line is going to be replaced in the wrong block (read source code instead of generated resurces)